src/transformers/training_args.py

```diff
@@ -1620,7 +1620,15 @@ def __post_init__(self):
             if not is_torch_available():
                 raise ValueError("lr_scheduler_type reduce_lr_on_plateau requires torch>=0.2.0")
 
+        use_badam = "badam_" in self.optim
+        if use_badam:
+            self.optim = self.optim.replace("badam_", "")
+
         self.optim = OptimizerNames(self.optim)
+
+        if use_badam:
+            self.optim = "badam_" + self.optim
```

Comment:
Having to remove and then add this back seems both convoluted and an indication that there's something funny about the implementation. Specifically, it seems like something which should be handled on the `OptimizerNames` side